internal/resource.go

```diff
@@ -63,13 +63,32 @@ func getUrl(u string, fileName string, ctx context.Context) (string, error) {
 	return fileName, nil
 }
 
-// GetUrlToDir downloads the given resource to the given directory and returns the path to it.
-func GetUrlToDir(u string, targetDir string, ctx context.Context) (string, error) {
+// GetUrlToDir downloads the given resource to a temporary file and returns the path to it.
+// Modify the GetUrlToDir function to ensure proper file cleanup
+func GetUrlToDir(u string, targetDir string, ctx context.Context) 
+(string, error) {
 	// create temporary name in the target directory.
 	h := sha256.New()
 	h.Write([]byte(u))
 	fileName := filepath.Join(targetDir, fmt.Sprintf(".%s", hex.EncodeToString(h.Sum(nil))))
-	return getUrl(u, fileName, ctx)
+
+	// Remove existing file if present
```

Comment:
We're using a random filename so the file is sort of guaranteed not to be present... so I think we don't need this.